v4/newrelic/segments_test.go

```diff
@@ -374,3 +376,24 @@ func TestSegmentsOnEmptyTransaction(t *testing.T) {
 	seg.End()
 	txn.End()
 }
+
+func TestStartExternalSegment(t *testing.T) {
+	app := newTestApp(t)
+	txn := app.StartTransaction("transaction")
+
+	req, _ := http.NewRequest("GET", "http://request.com/", nil)
+	seg1 := StartExternalSegment(txn, req)
```

Comment:
We should probably also include a test that has the txn in the req's context. However, that test will currently fail because we haven't yet implemented `FromContext`.

Reply:
I added a test, the test conditions being commented out (as they currently fail). It can be activated once context support is merged.